SitMe/Controllers/HomeController.cs

```diff
@@ -7,55 +8,61 @@
 using Microsoft.Extensions.Logging;
 using SitMe.Models;
 
-namespace SitMe.Controllers {
+namespace SitMe.Controllers
+{
     public class HomeController : Controller
     {
         private readonly ILogger<HomeController> _logger;
         private readonly IRepository<User> _userRepository;
         private readonly IRestaurantRepository _restaurantRepository;
 
-        public HomeController(  ILogger<HomeController> logger,
-                                IRepository<User> userRepository,
-                                IRestaurantRepository restaurantRepository)
+        public HomeController(
+            ILogger<HomeController> logger,
+            IRepository<User> userRepository,
+            IRestaurantRepository restaurantRepository)
         {
-            _logger = logger;
-            _userRepository = userRepository;
-            _restaurantRepository = restaurantRepository;
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            _userRepository = userRepository ?? throw new ArgumentNullException(nameof(userRepository));
+            _restaurantRepository = restaurantRepository ?? throw new ArgumentNullException(nameof(restaurantRepository));
         }
 
         public IActionResult Index()
         {
             return View();
         }
 
-        public IActionResult RestaurantProfile() {
+        public IActionResult RestaurantProfile()
+        {
             return View();
         }
 
-        public async Task<IActionResult> RestaurantList() {
-            // many to many query
-            //var restaurants = await _restaurantRepository.GetRetaurantsWithTags("polish");
-            var restaurants = await _restaurantRepository.GetRetaurantsWithTags(0, 9);
+        public async Task<IActionResult> RestaurantList()
+        {
+            var restaurants = await _restaurantRepository.GetRetaurantsWithTagsAsync(0, 9);
 
             return View(restaurants);
         }
 
-        public async Task<IActionResult> RestaurantListFilter(string filterByTest) {
-            var restaurants = await _restaurantRepository.GetRetaurantsWithTags(filterByTest);
+        public async Task<IActionResult> RestaurantListFilter(string filterByTest)
+        {
+            var restaurants = await _restaurantRepository.GetRetaurantsWithTagsAsync(filterByTest);
```

Comment:
Z tego co widzę, jest to dość prosta apka ale jeśli logika będzie się rozrastać przemyśl dodanie kolejnej warstwy, np `RestaruantService` i tam wstrzykiwuj repozytoria i obrabiaj dane. Narazie jest ok, bo nie ma logiki